### PR TITLE
MGMT-19125: Migrate prow-jobs-scraper to Konflux

### PR DESCRIPTION
**Pull request by giladravid16**
We want to change app-interface to deploy the images that Konflux builds and by default Konflux uses the full commit sha as the tag.
In order to avoid issues with telling app-interface to use the full commit sha, I am updating build_deploy.sh to push both the full sha and the short sha as image tags.
Also added licenses scraping.

Part of [MGMT-19125](https://issues.redhat.com//browse/MGMT-19125)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ dependencies = [
     "pandas==2.2.3",
     "numpy==2.1.2",
     "mmh3==5.0.1",
+    "pip-licenses",
```
Comment (adriengentil) [inline pyproject.toml:31]: I don't think we shoud add it here as it is not a direct dependency on the project. I would add it in the `RUN` command of the dockerfile instead

File: build_deploy.sh [hunk 2]
```diff
@@ -15,19 +15,25 @@ export CONTAINER_BUILD_EXTRA_PARAMS=${CONTAINER_BUILD_EXTRA_PARAMS:-"--no-cache"
...
 # Setup credentials to image registry
-${CONTAINER_CMD} login -u="${QUAY_USER}" -p="${QUAY_TOKEN}" quay.io
+#${CONTAINER_CMD} login -u="${QUAY_USER}" -p="${QUAY_TOKEN}" quay.io
 
 # Build and push latest image
 make build-image
 
 PROW_JOBS_SCRAPER_IMAGE_COMMIT_SHA="${PROW_JOBS_SCRAPER_IMAGE}:${PROW_JOBS_SCRAPER_TAG}"
-${CONTAINER_CMD} push "${PROW_JOBS_SCRAPER_IMAGE_COMMIT_SHA}"
+echo ${CONTAINER_CMD} push "${PROW_JOBS_SCRAPER_IMAGE_COMMIT_SHA}"
```
Comment (adriengentil) [inline build_deploy.sh:28]: the echo will just print, and it won't actually not push the tag

Comment (giladravid16) [reply build_deploy.sh:28]: my bad, I wanted to check that the images look fine and forget to remove it

**Review by adriengentil — APPROVED** [review]

**Pull request merged**